tests/unit/browser/test_qutescheme.py

```diff
@@ -27,30 +28,72 @@
 from qutebrowser.utils import objreg
 
 
-Dates = collections.namedtuple('Dates', ['yesterday', 'today', 'tomorrow'])
+class TestJavascriptHandler:
+
+    """Test the qute://javascript endpoint."""
+
+    # Tuples of fake JS files and their content.
+    js_files = [
+        ('foo.js', "var a = 'foo';"),
+        ('bar.js', "var a = 'bar';"),
+    ]
+
+    @pytest.fixture(autouse=True)
+    def patch_read_file(self, monkeypatch):
+        """Patch utils.read_file to return few fake JS files."""
+        def _read_file(path, binary=False):
+            """Faked utils.read_file."""
+            assert not binary
+            for filename, content in self.js_files:
+                if path == os.path.join('javascript', filename):
+                    return content
+            raise IOError("File not found {}!".format(path))
+
+        monkeypatch.setattr('qutebrowser.utils.utils.read_file', _read_file)
+
+    @pytest.mark.parametrize("filename, content", js_files)
+    def test_qutejavascript(self, filename, content):
+        url = QUrl("qute://javascript/{}".format(filename))
+        _mimetype, data = qutescheme.qute_javascript(url)
+
+        assert data == content
+
+    def test_qutejavascript_error(self):
```

Comment:
This test seems kinda useless to me, as it doesn't really test what happens when the `OSError` is raised.

I think it'd be better to call `qutescheme.data_for_url` here and ensure we actually get a `QuteSchemeOSError` (i.e. the user will get an error page, not a crash).